### PR TITLE
Make Github url configurable, default to github.com

### PR DESCRIPTION
**Pull request by fabianschwarzfritz**
Hi,

I'm not sure if you are interested in the change, but if you want,
take a look at it 😄 

Makes the Github url configurable, so that the tool can be used
for Github enterprise instances as well. The Github url
configuration defaults to api.github.com.

Cheers,
Fabian

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -8,6 +8,10 @@ class Config {
     this._config = config
   }
 
+  get url() {
```
Comment (patrickhulce) [inline lib/config.js]: url is a bit generic, I initially interpreted it as the URL to the repo

wdyt about something like `githubApiBaseUrl` or `apiBaseUrl`

File: lib/github.js [hunk 2]
```diff
@@ -1,17 +1,16 @@
 const log = require('debug')('git3po:github')
 const request = require('superagent-bluebird-promise')
 
-const GITHUB_URL = 'https://api.github.com'
-
 class GitHubApi {
-  constructor(token, repo, dryRun) {
+  constructor(url, token, repo, dryRun) {
+    this._url = url
```
Comment (patrickhulce) [inline lib/github.js]: can totally keep it as url here though 👍

File: test/config.test.js [hunk 3]
```diff
@@ -21,4 +21,15 @@ describe('config.js', () => {
       expect(config.startAt).to.eql(new Date('2017-01-14'))
     })
   })
+
+  describe('#url', () => {
```
Comment (patrickhulce) [inline test/config.test.js]: ace 👌

**Comment by fabianschwarzfritz** [comment]
Hi @patrickhulce , I've updated the pr with the proposed changed. Do you want me to squash the commits or do you prefer to keep the history? Cheers, Fabian

**Comment by patrickhulce** [comment]
I generally do the squash and merge if it's all the same to you. I updated the field in the yaml as well if that's alright with you! Looks good to me thanks again!

**Pull request merged**